cmd/service_linux.go

```diff
@@ -0,0 +1,335 @@
+		if err != nil {
+			return fmt.Errorf("failed to run 'systemctl daemon-reload' command due to following error: [%s]", err.Error())
+		}
+		return nil
+	},
+}
+
+//--------------- command section ends
+
+func newServiceFile(mountPath string, configPath string, serviceUser string) (string, error) {
+	serviceTemplate := ` [Unit]
+	Description=Cloudfuse is an open source project developed to provide a virtual filesystem backed by S3 or Azure storage.
+	After=network-online.target
+	Requires=network-online.target
+
+	[Service]
+	# User service will run as.
+	User={{.ServiceUser}}
+	# Path to the location Cloudfuse will mount to. Note this folder must currently exist.
+	Environment=MountingPoint={{.MountPath}}
+	# Path to the configuration file.
+	Environment=ConfigFile={{.ConfigFile}}
+
+	# Under the hood
+	Type=forking
+	ExecStart=/usr/bin/cloudfuse mount ${MountingPoint} --config-file=${ConfigFile} -o allow_other
+	ExecStop=/usr/bin/fusermount -u ${MountingPoint} -z
+
+	[Install]
+	WantedBy=multi-user.target
+	`
+
+	config := serviceOptions{
+		ConfigFile:  configPath,
+		MountPath:   mountPath,
+		ServiceUser: serviceUser,
+	}
+
+	tmpl, err := template.New("service").Parse(serviceTemplate)
+	if err != nil {
+		fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	folderList := strings.Split(mountPath, "/")
+	serviceName := folderList[len(folderList)-1] + ".service"
+	newFile, err := os.Create("/etc/systemd/system/" + serviceName)
+	if err != nil {
+		return "", fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	err = tmpl.Execute(newFile, config)
+	if err != nil {
+		return "", fmt.Errorf("error creating new service file: [%s]", err.Error())
+	}
+
+	return serviceName, nil
+}
+
+func setUser(serviceUser string, mountPath string, configPath string) error {
+
+	configFileInfo, err := os.Stat(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to stat file: %v", err)
+	}
+
+	// Get file's group ID
+	stat, ok := configFileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return fmt.Errorf("failed to get file system stats")
+	}
+	configGroupID := stat.Gid
+
+	// Get configFileGroup name
+	configFileGroup, err := user.LookupGroupId(fmt.Sprint(configGroupID))
+	if err != nil {
+		return fmt.Errorf("failed to lookup group: %v", err)
+	}
+
+	mountPathInfo, err := os.Stat(mountPath)
+	if err != nil {
+		return fmt.Errorf("failed to stat file: %v", err)
+	}
+
+	// Get file's group ID
+	stat, ok = mountPathInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return fmt.Errorf("failed to get file system stats")
+	}
+	mountGroupID := stat.Gid
+
+	// Get configFileGroup name
+	mountPathGroup, err := user.LookupGroupId(fmt.Sprint(mountGroupID))
+	if err != nil {
+		return fmt.Errorf("failed to lookup group: %v", err)
+	}
+
+	_, err = user.Lookup(serviceUser)
+	if err != nil {
+		if strings.Contains(err.Error(), "unknown user") {
+			//create the user
+			userAddCmd := exec.Command("sudo", "useradd", "-m", serviceUser)
```

Comment:
I think you should user "-r" instead of "-m" since "-m" will create a user that has a home directory and we probably don't want to do that.